pkg/edgeview/dispatcher/wss-server.go

```diff
@@ -0,0 +1,265 @@
+package main
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/gorilla/websocket"
+)
+
+type endPoint struct {
+	hostname  string
+	wsConn    *websocket.Conn
+}
+
+const (
+	noDeviceMsg string = "no device online\n+++Done+++"
+	tokenReqMsg string = "token is required"
+	moretwoMsg  string = "can't have more than 2 peers"
+)
+
+var upgrader = websocket.Upgrader{} // use default options
+// reqAddrTokeConn indexed by 'toke' then 'remoteAddr' strings
+var reqAddrTokenEP   map[string]map[string]endPoint
+// mutex for access the maps
+var connMutex sync.Mutex
+// connection id, keep inc
+var connID int
+// debug set
+var needDebug bool
+
+func socketHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
Can you please describe in comments how the hand shake happens between edge-node and edge-view client connections?

Reply:
will do.